src/include/platform/PersistedStorage.h

```diff
@@ -87,6 +91,139 @@ CHIP_ERROR Read(Key aKey, uint32_t & aValue);
+     *
+     * @param[in]     key               The name of the key in which to get.
+     * @param[in,out] buffer            A buffer to read the value into.
+     * @param[in]     buffer_size       The size of the buffer in bytes.
+     * @param[in,out] read_bytes_size   The number of bytes which were
+     *                                  copied into the buffer. Optionally can
+     *                                  provide nullptr if not needed.
+     * @param[in]     offset_bytes      The offset byte index in which to start
+     *                                  the read.
+     *
+     * @return CHIP_NO_ERROR the entry was successfully read
+     *         CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND the key is not
+     *                                                       present in the KVS
+     *         CHIP_ERROR_INTEGRITY_CHECK_FAILED found the entry, but the data
+     *                                           was corrupted
+     *         CHIP_ERROR_BUFFER_TOO_SMALL the buffer could not fit the entire
+     *                                     value, but as many bytes as possible
+     *                                     were written to it
+     *         CHIP_ERROR_WELL_UNINITIALIZED the KVS is not initialized
+     *         CHIP_ERROR_INVALID_ARGUMENT key is empty or too long or value is
+     *                                     too large
+     */
+    virtual CHIP_ERROR Get(const char * key, void * buffer, size_t buffer_size, size_t * read_bytes_size = nullptr,
+                           size_t offset_bytes = 0) const = 0;
+
+    /**
+     * @brief
+     * This overload of Get accepts a pointer to a trivially copyable object.
+     * The size of the object is inferred from the type.
+     *
+     * @param[in]      key     The name of the key in which to get.
+     * @param[in,out]  value   Pointer to a trivially copyable object.
+     *
+     * @return CHIP_NO_ERROR the entry was successfully read
+     *         CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND the key is not
+     *                                                      present in the KVS
+     *         CHIP_ERROR_INTEGRITY_CHECK_FAILED found the entry, but the data
+     *                                           was corrupted
+     *         CHIP_ERROR_BUFFER_TOO_SMALL the buffer could not fit the entire
+     *                                     value, but as many bytes as possible
+     *                                     were written to it
+     *         CHIP_ERROR_WELL_UNINITIALIZED the KVS is not initialized
+     *         CHIP_ERROR_INVALID_ARGUMENT key is empty or too long or value is
+     *                                     too large
+     */
+    template <typename T>
+    CHIP_ERROR Get(const char * key, T * value)
+    {
+        static_assert(std::is_trivially_copyable<T>(), "KVS values must copyable");
+        static_assert(!std::is_pointer<T>(), "KVS values cannot be pointers");
+        return Get(key, value, sizeof(T));
```

Comment:
Please `static_assert(CHAR_BIT == 8)`, since this code is assuming that.  Otherwise we have to carefully check for bytes vs octets everywhere here...

Reply:
Done